src/Message/CompletePurchaseRequest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace DigiTickets\OmnipayOptomanyCheckout\Message;
+
+use DNAPayments\DNAPayments;
+
+class CompletePurchaseRequest extends AbstractRequest
+{
+    public function getData()
+    {
+        $data = [];
+
+        return $data;
+    }
+
+    public function sendData($data)
+    {
+        $postJson = file_get_contents('php://input');
+
+        $postData = empty($data) ? json_decode($postJson, true) : $data;
+
+        if (empty($postData)) {
+            throw new \Exception("No data in callback");
```

Comment:
Please use Omnipay\Common\Exception\InvalidRequestException for invalid data (same below)